app/services/report_scanner/puppet_report_scanner.rb

```diff
@@ -0,0 +1,14 @@
+module Foreman
+  class PuppetReportScanner
+    class << self
+      def scan(report)
+        report.origin = 'Puppet' if puppet_report? report
+      end
+
+      def puppet_report?(report)
+        first_log = report.logs.first
+        first_log && first_log.source.value == 'Puppet'
```

Comment:
this does not work for me, the source I see locally is e.g. "//scapclient.example.tst/Puppet", we should perhaps use `=~` instead?